lxd/devlxd_util.go

```diff
@@ -63,3 +68,101 @@ func NewRequestWithContext(ctx context.Context, method string, url string, data
 
 	return req, nil
 }
+
+// responseCapture is a custom http.ResponseWriter that captures the response
+// header, body, and status code for later inspection.
+type responseCapture struct {
+	header     http.Header
+	body       *bytes.Buffer
+	statusCode int
+	url        url.URL
+}
+
+// NewResponseCapture creates a new responseCapture for the given request.
+func NewResponseCapture(req *http.Request) *responseCapture {
```

Comment:
Can these go into the `response` package, they dont appear to be devlxd specific?